deque/iterator.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package deque
+
+import (
+	"container/list"
+	"reflect"
+)
+
+// Iterator allows in order iteration of the deque elements.
+// The iterator is valid as long as no elements are added or removed from the deque.
+type Iterator struct {
+	deque *Deque
+	elem  *list.Element
+	block blockT
+	index int
+	pos   int
+}
+
+// Iterator returns an `Iterator` for in order iteration of the elements.
+func (d *Deque) Iterator() Iterator {
+	front := d.blocks.Front()
+	return Iterator{
+		deque: d,
+		elem:  front,
+		block: front.Value.(blockT),
+		pos:   d.frontIdx - 1,
+	}
+}
+
+// Next returns true if there is a value, and the value is populated with the
```

Comment:
please update the comment to say something like
value must be a pointer to the type stored in the deque.